File: myportal/frontend/src/agp-kasse.js
import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-grid/src/vaadin-grid.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';
import './agp-menu.js';

class AgpKasse extends PolymerElement {

    static get template() {
        return html`
<style include="shared-styles">
                :host {
                    display: block;
                    height: 100%;
                }
            </style>
<vaadin-vertical-layout style="width: 100%; height: 100%;" id="vaadinVerticalLayout">
 <agp-menu id="agpMenu"></agp-menu>
 <label id="labelRechnungsnummer">Rechnungsnummer:</label>
 <label id="labelMatrikelnummer">Matrikelnummer:</label>
 <label id="labelName">Name:</label>
 <label id="labelArtikelliste">Artikelliste:</label>
 <vaadin-grid id="gridArtikelliste"></vaadin-grid>
 <label id="labelGesamtpreis">Gesamtpreis:</label>
 <vaadin-button theme="tertiary" id="buttonBestellen">
  Verbindlich bestellen
 </vaadin-button>
</vaadin-vertical-layout>
`;
    }

    static get is() {
        return 'agp-kasse';
    }

    static get properties() {
        return {
            // Declare your properties here.
        };
    }
}

customElements.define(AgpKasse.is, AgpKasse);

File: myportal/src/main/java/swp/portal/AgpKasse.java
package swp.portal;

import com.vaadin.flow.templatemodel.TemplateModel;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.grid.Grid;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.component.polymertemplate.Id;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.router.Route;

/**
 * A Designer generated component for the agp-kasse template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("agp-kasse")
@JsModule("./src/agp-kasse.js")
public class AgpKasse extends PolymerTemplate<AgpKasse.AgpKasseModel> {

	
	@Id("vaadinVerticalLayout")
	private VerticalLayout vaadinVerticalLayout;
	@Id("labelRechnungsnummer")
	private Label labelRechnungsnummer;
	@Id("labelMatrikelnummer")
	private Label labelMatrikelnummer;
	@Id("labelName")
	private Label labelName;
	@Id("labelArtikelliste")
	private Label labelArtikelliste;
	@Id("gridArtikelliste")
	private Grid gridArtikelliste;
	@Id("labelGesamtpreis")
	private Label labelGesamtpreis;
	@Id("buttonBestellen")
	private Button buttonBestellen;
	@Id("labelUeberweisung")
	private Label labelUeberweisung;
	
	
    /**
     * Creates a new AgpKasse.
     */
    public AgpKasse() {
        // You can initialise any data required for the connected UI components here.
    }

    /**
     * This model binds properties between AgpKasse and agp-kasse
     */
    public interface AgpKasseModel extends TemplateModel {
        // Add setters and getters for template properties here.
    }
}
